plugin-lifecycle/src/main/java/com/mapbox/maps/plugin/lifecycle/MapboxLifecyclePluginImpl.kt

```diff
@@ -10,58 +10,65 @@ import android.content.ComponentCallbacks2.TRIM_MEMORY_RUNNING_MODERATE
 import android.content.ComponentCallbacks2.TRIM_MEMORY_UI_HIDDEN
 import android.content.res.Configuration
 import android.view.View
-import androidx.lifecycle.Lifecycle
-import androidx.lifecycle.LifecycleObserver
-import androidx.lifecycle.OnLifecycleEvent
-import androidx.lifecycle.ViewTreeLifecycleOwner
+import androidx.lifecycle.*
 import com.mapbox.maps.MapboxLifecycleObserver
-import com.mapbox.maps.logE
 import com.mapbox.maps.logW
 import com.mapbox.maps.plugin.Plugin.Companion.MAPBOX_LIFECYCLE_PLUGIN_ID
 import com.mapbox.maps.plugin.delegates.MapPluginProviderDelegate
 
 /**
  * Concrete implementation of MapboxLifecyclePlugin.
  */
-class MapboxLifecyclePluginImpl : MapboxLifecyclePlugin {
+class MapboxLifecyclePluginImpl constructor(mapView: View?) : MapboxLifecyclePlugin,
+  LifecycleOwner {
+  private lateinit var viewLifecycleRegistry: ViewLifecycleRegistry
+
+  init {
+    mapView?.let {
+      viewLifecycleRegistry = ViewLifecycleRegistry(
```

Comment:
If no subscriptions is made here - what is the point to init viewLifecycleRegistry?

Reply:
yeah, this is a commit that prototypes Lukas's idea of initialising the viewLifecycleRegistry in constructor, however given the above conversation, we'd probably not initialising it in the constructor.